plugins/flytekit-polars/flytekitplugins/polars/sd_transformers.py

```diff
@@ -40,23 +40,32 @@ def encode(
         structured_dataset_type: StructuredDatasetType,
     ) -> literals.StructuredDataset:
         df = typing.cast(pl.DataFrame, structured_dataset.dataframe)
-
+        random_string = ctx.file_access.get_random_string()
         local_dir = ctx.file_access.get_random_local_directory()
-        local_path = f"{local_dir}/00000"
-
+        local_path = ctx.file_access.join(
+            local_dir,
+            random_string,
+        )
         # Polars 0.13.12 deprecated to_parquet in favor of write_parquet
         if hasattr(df, "write_parquet"):
             df.write_parquet(local_path)
         else:
             df.to_parquet(local_path)
-        remote_dir = typing.cast(str, structured_dataset.uri)
-        if not remote_dir:
-            remote_dir = ctx.file_access.join(
-                ctx.file_access.raw_output_prefix,
-                ctx.file_access.get_random_string(),
+
+        if structured_dataset.uri is not None:
+            remote_dir = structured_dataset.uri
+            sd_uri = ctx.file_access.join(
+                remote_dir,
+                random_string,
+            )
+        else:
+            remote_dir = ctx.file_access.get_random_remote_directory()
+            sd_uri = ctx.file_access.join(
+                remote_dir,
+                random_string,
             )
         ctx.file_access.upload_directory(local_dir, remote_dir)
```

Comment:
i actually want to get rid of this dual writes, one to local directory and then to remote. can we simply write to remote as a stream?

Reply:
I've updated the code to write to a BytesIO stream. Let me know if that is in line with what you were thinking!